src/main/java/org/jsoup/select/QueryParser.java

```diff
@@ -149,10 +150,19 @@ else if (tq.matches("["))
             else if (tq.matchesAny(combinators))
                 if (sq.length() > 0)
                     break;
-                else
-                    tq.consume();
-            else
+                else{
```

Comment:
I don't get why we're popping it off and then re-adding. Can't it just happen in one step? Seems kludgy